ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+            feature_support = get(self._platform_settings, 'feature_support')
+            queue_monitor_length_notify = ""
+            if feature_support is not None:
+                queue_monitor_length_notify = get(feature_support, 'queue_monitor_length_notify')
+            if (
+                    queue_monitor_length_notifying is not None and
+                    queue_monitor_length_notify is not False
+            ):
+                queue_monitor_length_dict.update({'notifying': queue_monitor_length_notifying})
+            if get(queue_monitor_length, 'log') is not None:
+                queue_monitor_length_dict.update({'log': get(queue_monitor_length, 'log')})
+            return queue_monitor_length_dict
+        return None
+
+    @cached_property
+    def name_server(self):
+        name_servers = get(self._hostvars, 'name_servers')
+        if name_servers is not None:
+            return {
+                'source':
+                    {
+                        'vrf': self._mgmt_interface_vrf
+                    },
+                'nodes': name_servers
+            }
+        return None
+
+    @cached_property
+    def redundancy(self):
+        redundancy = get(self._hostvars, 'redundancy')
+        if redundancy is not None:
+            protocol = get(redundancy, 'protocol')
+            if protocol is not None:
+                return {'protocol': protocol}
+        return None
+
+    @cached_property
+    def snmp_server(self):
+        snmp_settings = get(self._hostvars, 'snmp_settings')
+        if snmp_settings is not None:
+            snmp_server = {}
+            compute_local_engineid = get(snmp_settings, 'compute_local_engineid')
+            if compute_local_engineid is True:
+                local_engine_id = sha1(f"{self._hostname}{self._mgmt_ip}".encode('utf-8')).hexdigest()
+                snmp_server.update({'engineid': {'local': local_engine_id}})
+            contact = get(snmp_settings, 'contact')
+            if contact is not None:
+                snmp_server.update({'contact': contact})
+            if get(snmp_settings, 'location') is not None:
+                snmp_location = get(self._hostvars, 'fabric_name')
+                dc_name = get(self._hostvars, 'dc_name')
+                if dc_name is not None:
+                    snmp_location = f"{snmp_location} {dc_name}"
+                pod_name = get(self._hostvars, 'pod_name')
+                if pod_name is not None:
+                    snmp_location = f"{snmp_location} {pod_name}"
+                rack = get(self._switch_data, 'rack')
+                if rack is not None:
+                    snmp_location = f"{snmp_location} {rack}"
+                snmp_location = f"{snmp_location} {self._hostname}"
+                snmp_server.update({'location': snmp_location})
+            users = get(snmp_settings, 'users')
+            if users is not None:
+                snmp_server.update({'users': []})
+                for user in users:
```

Comment:
lets do something like this:
```
users_dict = {}
for user in users:
  ... current for loop
if users_dict:
  snmp_server["users"] = users_dict
```


Additionally the for loop can be simplified - there were some things written only because of Jinja